proto/ibc/applications/fee/v1/tx.proto

```diff
@@ -0,0 +1,73 @@
+syntax = "proto3";
+
+package ibc.applications.fee.v1;
+
+import "gogoproto/gogo.proto";
+import "ibc/applications/fee/v1/fee.proto";
+import "ibc/core/channel/v1/channel.proto";
+option go_package = "github.com/cosmos/ibc-go/v3/modules/apps/29-fee/types";
+
+// Msg defines the ibc/fee Msg service.
+service Msg {
+  // RegisterCounterpartyAddress defines a rpc handler method for MsgRegisterCounterpartyAddress
+  // RegisterCounterpartyAddress is called by the relayer on each channelEnd and allows them to specify their
+  // counterparty address before relaying. This ensures they will be properly compensated for forward relaying since
+  // destination chain must send back relayer's source address (counterparty address) in acknowledgement. This function
+  // may be called more than once by a relayer, in which case, latest counterparty address is always used.
+  rpc RegisterCounterpartyAddress(MsgRegisterCounterpartyAddress) returns (MsgRegisterCounterpartyAddressResponse);
+  // PayPacketFee defines a rpc handler method for MsgPayPacketFee
+  // PayPacketFee is an open callback that may be called by any module/user that wishes to escrow funds in order to
+  // incentivize the relaying of the packet at the next sequence
+  rpc PayPacketFee(MsgPayPacketFee) returns (MsgPayPacketFeeResponse);
+  // PayPacketFeeAsync defines a rpc handler method for MsgPayPacketFeeAsync
+  // PayPacketFeeAsync is an open callback that may be called by any module/user that wishes to escrow funds in order to
+  // incentivize the relaying of a known packet
+  rpc PayPacketFeeAsync(MsgPayPacketFeeAsync) returns (MsgPayPacketFeeAsyncResponse);
+}
+
+// MsgRegisterCounterpartyAddress is the request type for registering the counter party address
+message MsgRegisterCounterpartyAddress {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string address              = 1;
+  string counterparty_address = 2 [(gogoproto.moretags) = "yaml:\"counterparty_address\""];
+}
+
+// MsgRegisterCounterpartyAddressResponse defines the Msg/RegisterCounterypartyAddress response type
+message MsgRegisterCounterpartyAddressResponse {}
+
+// MsgPayPacketFee defines the request type PayPacketFee RPC
+// This Msg can be used to pay for a packet at the next sequence send & should be combined with the Msg that will be
+// paid for
+message MsgPayPacketFee {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  ibc.applications.fee.v1.Fee fee = 1 [(gogoproto.nullable) = false];
+  // source channel port identifier
+  string source_port_id = 2 [(gogoproto.moretags) = "yaml:\"source_port_id\""];
+  // source channel unique identifier
+  string source_channel_id = 3 [(gogoproto.moretags) = "yaml:\"source_channel_id\""];
+  // account address to refund fee if necessary
+  string          signer   = 4;
```

Comment:
Is this field being validated anywhere that it is the same signer as the one that signed the transaction? If not, anyone will be able to escrow fees of any amount on behalf of anyone else.

Reply:
Found them being handled in `sdk.Msg.GetSigners`. This can be disregarded.